### PR TITLE
fix: squad card weird look

### PR DESCRIPTION
**Pull request by sshanzel**
## Changes
- The reason why the cards looked weird was, it was missing the border color, header image, and alignment of the elements.
- It was aliased to a different property to fit the previous card. But since we don't need the previous card anymore, we don't have to alias it. Also to avoid future confusion that it is actually part of the schema.

Preview:

Screenshot 2024-09-12 at 2 01 50 PM

## Events

Did you introduce any new tracking events?

## Experiment

Did you introduce any new experiments?

## Manual Testing

> [!CAUTION]
> Please make sure existing components are not breaking/affected by this PR

MI-526

### Preview domain
https://mi-526.preview.app.daily.dev

### CODE REVIEW

File: packages/shared/src/components/cards/squad/SquadList.tsx [hunk 1]
```diff
@@ -62,7 +62,7 @@ export const SquadList = ({
         copy={{ join: 'Join', view: 'View' }}
         data-testid="squad-action"
         showViewSquad
-        buttonVariants={[ButtonVariant.Secondary, ButtonVariant.Secondary]}
+        buttonVariants={[ButtonVariant.Secondary, ButtonVariant.Float]}
```
Comment (sshanzel) [inline packages/shared/src/components/cards/squad/SquadList.tsx:65]: View buttons should be Float.

image

File: packages/shared/src/graphql/sources.ts [hunk 2]
```diff
@@ -62,8 +62,6 @@ export interface Squad extends Source {
   memberPostingRole: SourceMemberRole;
   memberInviteRole: SourceMemberRole;
   referralUrl?: string;
-  banner?: string;
-  borderColor?: string;
```
Comment (sshanzel) [inline packages/shared/src/graphql/sources.ts:66]: Kinda confusing to have this when it is not really part of the schema. Maybe we should extend this base interface if we want to modify its properties that are not found in the actual schema. Just my thoughts.

**Review by nensidosari — APPROVED** [review]

File: packages/shared/src/components/cards/squad/SquadGrid.tsx [hunk 3]
```diff
@@ -61,12 +72,12 @@ export const SquadGrid = ({
...
         <Image
           className="h-full w-full object-cover"
-          src={banner || cloudinary.squads.directory.cardBannerDefault}
+          src={headerImage || cloudinary.squads.directory.cardBannerDefault}
           alt="Banner image for source"
         />
       </div>
       <div className="-mt-12 flex flex-1 flex-col rounded-t-16 bg-background-subtle p-4">
-        <div className="mb-3 flex items-end justify-between">
+        <div className="-mt-14 mb-3 flex items-end justify-between">
```
[the remark this answers is not in the document]
Comment (sshanzel) [reply packages/shared/src/components/cards/squad/SquadGrid.tsx:80]: Oh, didn't notice that. Agree, it kinda feels hacky. Let me check if we can restructure it a bit 👀

File: packages/shared/src/components/cards/squad/SquadGrid.tsx [hunk 4]
```diff
@@ -97,11 +108,15 @@ export const SquadGrid = ({
...
-            className={{ button: '!btn-secondary z-0 w-full' }}
+            className={{ button: 'z-0 w-full' }}
             squad={source}
             origin={Origin.SquadDirectory}
             onSuccess={() => router.push(source.permalink)}
             data-testid="squad-action"
+            buttonVariants={[
+              ButtonVariant.Secondary,
+              isMobile ? ButtonVariant.Float : ButtonVariant.Secondary,
+            ]}
```
[the remark this answers is not in the document]
Comment (sshanzel) [reply packages/shared/src/components/cards/squad/SquadGrid.tsx:119]: Yeah, feels like these things need to be consistent across the board. I've sent a message to the design team earlier today. I will send a reply here once they respond 👀

[the remark this answers is not in the document]
Comment (sshanzel) [reply packages/shared/src/components/cards/squad/SquadGrid.tsx:117]: The classes for `w-full` on this one should be unique as the other cards have small buttons. For the variants, the change was based from [this discussion](https://dailydotdev.slack.com/archives/C069XNYMDGV/p1725364222055539?thread_ts=1725358287.600279&cid=C069XNYMDGV), if they would want to make the behavior consistent, we shall open another PR 🙏

File: packages/shared/src/components/cards/squad/SquadGrid.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (sshanzel) [reply packages/shared/src/components/cards/squad/SquadGrid.tsx]: Should be fixed now. The header image was not fixed and needed to always move the elements. The position should now be absolute and then we move the other elements as necessary.

**Pull request merged**